ts/screens/authentication/cie/CiePinScreen.tsx

```diff
@@ -48,140 +56,163 @@ const styles = StyleSheet.create({
 });
 
 const CIE_PIN_LENGTH = 8;
+const FORGOT_PIN_PAGE_URL =
+  "https://www.cartaidentita.interno.gov.it/richiesta-di-ristampa/";
 
-/**
- * A screen that allow the user to insert the Cie PIN.
- */
-class CiePinScreen extends React.PureComponent<Props, State> {
-  private continueButtonRef = React.createRef<FooterWithButtons>();
-  private pinPadViewRef = React.createRef<View>();
-  constructor(props: Props) {
-    super(props);
-    this.state = { pin: "" };
-  }
+const CiePinScreen: React.FC<Props> = props => {
+  const { showAnimatedModal, hideModal } = useContext(LightModalContext);
+  const { navigate } = useContext(NavigationContext);
+
+  const onOpenForgotPINPage = () => openWebUrl(FORGOT_PIN_PAGE_URL);
 
-  private getContextualHelp = () => ({
+  const { present } = useIOBottomSheet(
+    <View>
+      <Body>{I18n.t("authentication.cie.pin.bottomSheetText")}</Body>
+      <Link onPress={onOpenForgotPINPage}>
+        {I18n.t("authentication.cie.pin.bottomSheetCTA")}
+      </Link>
+    </View>,
+    I18n.t("authentication.cie.pin.bottomSheetTitle"),
+    300
+  );
+
+  const getContextualHelp = () => ({
     title: I18n.t("authentication.cie.pin.contextualHelpTitle"),
     body: () => (
       <Markdown>{I18n.t("authentication.cie.pin.contextualHelpBody")}</Markdown>
     )
   });
 
-  private onProceedToCardReaderScreen = (url: string) => {
-    const ciePin = this.state.pin;
-    this.setState({ url, pin: "" }, () => {
-      this.props.navigation.navigate({
-        routeName: ROUTES.CIE_CARD_READER_SCREEN,
-        params: { ciePin, authorizationUri: url }
-      });
-      this.props.hideModal();
-    });
+  const [pin, setPin] = useState("");
+  const [url, setUrl] = useState<string | undefined>(undefined);
+
+  const continueButtonRef = useRef<FooterWithButtons>(null);
+  const PINPadViewRef = useRef<View>(null);
+
+  const onProceedToCardReaderScreen = async (url: string) => {
+    setPin("");
+    setUrl(url);
   };
 
-  private handleAuthenticationOverlayOnClose = () => {
-    // reset the pin if user abort process during loading
-    this.setState({ pin: "" }, this.props.hideModal);
+  // Handles URL receival.
+  useEffect(() => {
+    if (url === undefined) {
```

Comment:
Why use an effect to listen about the url changes?
Maybe we could avoid to use  `url/setUrl` state and handle it inside `onProceedToCardReaderScreen`
Actually we read this value only when it is created

```typescript
const onProceedToCardReaderScreen = async (url: string) => {
    setPin("");
    navigate({
      routeName: ROUTES.CIE_CARD_READER_SCREEN,
      params: {
        ciePin: pin,
        authorizationUri: url
      }
    });
    hideModal();
  };
```

Reply:
Missed this, the URL is used here and only here.